runtime/vm/classsupport.c

```diff
@@ -988,6 +1005,75 @@ arbitratedLoadClass(J9VMThread* vmThread, U_8* className, UDATA classNameLength,
 	return foundClass;
 }
 
+#if defined(J9VM_OPT_SNAPSHOTS)
+/* TODO: Revisit this function. */
+BOOLEAN
+loadWarmClassFromSnapshot(J9VMThread *vmThread, J9ClassLoader *classLoader, J9Class *clazz)
+{
+	BOOLEAN rc = FALSE;
+	BOOLEAN failed = FALSE;
+
+	if (J9_ARE_NO_BITS_SET(clazz->classFlags, J9ClassIsLoadedFromSnapshot)) {
+		J9JavaVM *vm = vmThread->javaVM;
+		J9Class *superClazz = clazz->superclasses[J9CLASS_DEPTH(clazz) - 1];
+		J9ITable *itable = (J9ITable *)clazz->iTable;
+		J9Class *interface = NULL;
+		const char *className = (const char *)J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
+
+		clazz->classFlags |= J9ClassIsLoadedFromSnapshot;
+
+		/* Load superclasses and interfaces first. */
+		if (NULL != superClazz) {
+			if (!loadWarmClassFromSnapshot(vmThread, classLoader, superClazz)) {
+				goto done;
+			}
+		}
+
+		while (NULL != itable) {
+			interface = itable->interfaceClass;
+			if (NULL != interface) {
+				if (!loadWarmClassFromSnapshot(vmThread, classLoader, interface)) {
+					goto done;
+				}
+			}
+			itable = itable->next;
+		}
+
+		initializeSnapshotJ9Class(vm, clazz);
+
+		if (J9_ARE_ALL_BITS_SET(vmThread->javaVM->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_CLASS_OBJECT_ASSIGNED)
+			&& (NULL == clazz->classObject)
+		) {
+			clazz = initializeSnapshotClassObject(vm, classLoader, clazz);
+		}
+
+		TRIGGER_J9HOOK_VM_INTERNAL_CLASS_LOAD(vm->hookInterface, vmThread, clazz, failed);
+		TRIGGER_J9HOOK_VM_CLASS_LOAD(vm->hookInterface, vmThread, clazz);
+
+		if (failed) {
+			Trc_VM_snapshot_loadWarmClassFromSnapshot_ClassLoadHookFailed(vmThread, clazz, className);
```

Comment:
This previously reloaded `className`: why was that removed?

Reply:
The `clazz->romClass` doesn't change. I don't think it is necessary to reload.

Reply:
It's not clear that's a fair assumption: the class-load hooks above might change where the name is. Please put the reload back (with a comment):
```c
			/* the class-load hooks above may necessitate reloading className */
			className = (const char *)J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
```